python-regression/util/transaction_bundle_logic/transaction_logic.py

```diff
@@ -25,26 +26,37 @@ def create_transaction_bundle(address, tag, value):
     return bundle
 
 
-def create_and_attach_transaction(api, arg_list):
+def create_and_attach_transaction(api, value_transaction, arg_list, *reference):
     """
     Create a transaction and attach it to the tangle.
 
     :param api: The api target you would like to make the call to
+    :param value_transaction: A bool to determine if this transaction is a value or zero value transaction
     :param arg_list: The argument list (dictionary) for the transaction
     :return sent: The return value for the attachToTangle call (contains the attached transaction trytes)
     """
     transaction = ProposedTransaction(**arg_list)
 
-    bundle = ProposedBundle()
-    bundle.add_transaction(transaction)
-    bundle.finalize()
-    trytes = str(bundle[0].as_tryte_string())
+    if value_transaction:
+        inputs = api.get_inputs(start=0, stop=10, threshold=0)
+        prepared_transaction = api.prepare_transfer(
+            transfers=[transaction],
+            inputs=[inputs['inputs'][0]],
+            change_address=Address(getattr(static, "TEST_EMPTY_ADDRESS"))
+        )
+    else:
+        prepared_transaction = api.prepare_transfer(
+            transfers=[transaction]
+        )
 
     gtta = api.get_transactions_to_approve(depth=3)
```

Comment:
just that you know, if you want to make sure you won't be getting tips that may conflict with your reference you can pass the reference as an arg to GTTA.

Not sure if this is relevant here